PowerUp/app/src/main/java/powerup/systers/com/utils/CSVReader.java

```diff
@@ -19,94 +20,53 @@
-
-        while ((line = br.readLine()) != null) {
-            String[] row = line.split(csvSplitBy);
-            clothList.add(new Clothes(Integer.valueOf(row[0]), row[1], Integer.valueOf(row[2]), 0));
-        }
-        br.close();
-        return clothList;
-    }
-
-    // read Hair from csv file & return a List of Hair data.
-    public List<Hair> getHairList() throws IOException {
-        List<Hair> hairList = new ArrayList<>();
-        BufferedReader br = null;
-        br = new BufferedReader(new InputStreamReader(context.getAssets().open("Hair.csv")));
-
-        while ((line = br.readLine()) != null) {
-            String[] row = line.split(csvSplitBy);
-            hairList.add(new Hair(Integer.valueOf(row[0]), row[1], Integer.valueOf(row[2]), 0));
-        }
-        br.close();
-        return hairList;
-    }
-
-    // read Accessories from csv file & return a List of Accessory data.
-    public List<Accessory> getAccessoriesList() throws IOException {
-        List<Accessory> accessoryList = new ArrayList<>();
-        BufferedReader br = null;
-        br = new BufferedReader(new InputStreamReader(context.getAssets().open("Accessories.csv")));
-
-        while ((line = br.readLine()) != null) {
-            String[] row = line.split(csvSplitBy);
-            accessoryList.add(new Accessory(Integer.valueOf(row[0]), row[1], Integer.valueOf(row[2]), 0));
+        try{
+            br =  new BufferedReader(new InputStreamReader(context.getAssets().open(filename)));
+            while ((line = br.readLine()) != null) {
+                row = line.split(csvSplitBy);
+                if(row.length > 0) {
+                    switch (type) {
+                        case 0: list.add((T) new Question(Integer.valueOf(row[0]), Integer.valueOf(row[1]), row[2]));
+                                break;
+                        case 1: list.add((T) new Answer(Integer.valueOf(row[0]), Integer.valueOf(row[1]),
+                                    row[2], Integer.valueOf(row[3]), Integer.valueOf(row[4])));
+                            break;
+                        case 2: list.add((T) new Scenario(Integer.valueOf(row[0]), row[1], row[2], row[3], Integer.valueOf(row[4]),
+                                    Integer.valueOf(row[5]), 0, Integer.valueOf(row[6]), 0));
+                                break;
+                        case 3: list.add((T) new Clothes(Integer.valueOf(row[0]), row[1], Integer.valueOf(row[2]), 0));
+                                break;
+                        case 4: list.add((T) new Hair(Integer.valueOf(row[0]), row[1], Integer.valueOf(row[2]), 0));
+                                break;
+                        case 5: list.add((T) new Accessory(Integer.valueOf(row[0]), row[1], Integer.valueOf(row[2]), 0));
+
+                    }
+                }
+            }
+        } catch (IOException e) {
+            Log.e(TAG, e.toString());
```

Comment:
IOException occurs at 2 different places, once opening and once at closing of same file name. I would suggest adding specific error message, e.g. "IOException while opening file + filename" and "IOException while closing file + filename". This would help in debugging case or Exception.